src/librustc/ty/layout.rs

```diff
@@ -548,28 +548,30 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
                         return Err(LayoutError::Unknown(ty));
                     }
                 }
+                if let Some(count) = count.assert_usize(tcx) {
```

Comment:
You should use `count.assert_usize(tcx).ok_or(LayoutError::Unknown(ty))?` to minimize the diff.